subsys/caf/modules/sensor_data_aggregator.c

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <drivers/sensor.h>
+#include <event_manager.h>
+
+#include <caf/events/sensor_event.h>
+#include <caf/events/sensor_aggregator_event.h>
+#include <caf/sensor_manager.h>
+
+#define MODULE sensor_data_aggregator
+#include <caf/events/module_state_event.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_CAF_SENSOR_MANAGER_LOG_LEVEL);
+
+#define DT_DRV_COMPAT caf_aggregator
+
+#define __DEFINE_DATA(i, agg_id, size)							\
+	__ALIGNED(4) static uint8_t agg_ ## agg_id ## _buff_ ## i ## _data[size];
```

Comment:
note - I am wondering if we shouldn't use one buffer that we split into subsections.
The only downside I see is the alignment (we won't be able to enforce it for subsections). But will this be a problem... if size of data is not aligned we will have alignment problems on the way anyway.  So maybe it's not something we should worry about now.

Reply:
Discussed offline. Solution worth mentioning, but not implementing right now.